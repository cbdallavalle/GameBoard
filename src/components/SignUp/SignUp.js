import React from 'react';
import PropTypes from 'prop-types';
import './SignUp.css';

export const SignUp = (props) => {
  const { 
    handleChange, 
    email, 
    passwordOne, 
    firstName, 
    lastName, 
    passwordTwo 
  } = props;

  return (
    <div className="SignUp">
      <input 
        type="text"
        placeholder="first name"
        name="firstName"
        value={ firstName }
        onChange={ (event) => handleChange(event) }
      />      
      <input 
        type="text"
        placeholder="last name"
        name="lastName"
        value={ lastName }
        onChange={ (event) => handleChange(event) }
      />
      <input 
        type="email"
        placeholder="email"
        name="email"
        onChange={ (event) => handleChange(event) }
        value={ email }
      />
      <input 
        type="password"
        placeholder="password"
        name="passwordOne"
        onChange={ (event) => handleChange(event) }
        value={ passwordOne }
      />
      <input 
        type="password"
        placeholder="retype password"
        name="passwordTwo"
        onChange={ (event) => handleChange(event) }
        value={ passwordTwo }
      />
    </div>
  );
};

SignUp.propTypes = {
  handleChange: PropTypes.func.isRequired,
  email: PropTypes.string.isRequired,
  passwordOne: PropTypes.string.isRequired,
  passwordTwo: PropTypes.string.isRequired,
  firstName: PropTypes.string.isRequired,
  lastName: PropTypes.string.isRequired
};