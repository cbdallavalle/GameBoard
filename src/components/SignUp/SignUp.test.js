import React from 'react';
import { shallow } from 'enzyme';
import { SignUp } from './SignUp';

describe("SignUp", () => {
  const mockHandleChange = jest.fn();
  const props = {email: 'email', passwordOne: 'password', passwordTwo: 'password', firstName: 'wren', lastName: 'little' }

  it('should match the snapshot', () => {
    const wrapper = shallow(
      <SignUp 
        handleChange={mockHandleChange}
        {...props}
      />
    )

    expect(wrapper).toMatchSnapshot();
  })
})