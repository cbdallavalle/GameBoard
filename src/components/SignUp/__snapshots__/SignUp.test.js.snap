// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignUp should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    email="email"
    handleChange={[Function]}
    name="wren"
    passwordOne="password"
    passwordTwo="password"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <input
          className="login-input"
          name="name"
          onChange={[Function]}
          placeholder="Name"
          type="text"
          value="wren"
/>,
        <input
          className="login-input"
          name="email"
          onChange={[Function]}
          placeholder="email"
          type="email"
          value="email"
/>,
        <input
          className="login-input"
          name="passwordOne"
          onChange={[Function]}
          placeholder="Password"
          type="password"
          value="password"
/>,
        <input
          className="login-input"
          name="passwordTwo"
          onChange={[Function]}
          placeholder="Retype password"
          type="password"
          value="password"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "login-input",
          "name": "name",
          "onChange": [Function],
          "placeholder": "Name",
          "type": "text",
          "value": "wren",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "login-input",
          "name": "email",
          "onChange": [Function],
          "placeholder": "email",
          "type": "email",
          "value": "email",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "login-input",
          "name": "passwordOne",
          "onChange": [Function],
          "placeholder": "Password",
          "type": "password",
          "value": "password",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "login-input",
          "name": "passwordTwo",
          "onChange": [Function],
          "placeholder": "Retype password",
          "type": "password",
          "value": "password",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <input
            className="login-input"
            name="name"
            onChange={[Function]}
            placeholder="Name"
            type="text"
            value="wren"
/>,
          <input
            className="login-input"
            name="email"
            onChange={[Function]}
            placeholder="email"
            type="email"
            value="email"
/>,
          <input
            className="login-input"
            name="passwordOne"
            onChange={[Function]}
            placeholder="Password"
            type="password"
            value="password"
/>,
          <input
            className="login-input"
            name="passwordTwo"
            onChange={[Function]}
            placeholder="Retype password"
            type="password"
            value="password"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "login-input",
            "name": "name",
            "onChange": [Function],
            "placeholder": "Name",
            "type": "text",
            "value": "wren",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "login-input",
            "name": "email",
            "onChange": [Function],
            "placeholder": "email",
            "type": "email",
            "value": "email",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "login-input",
            "name": "passwordOne",
            "onChange": [Function],
            "placeholder": "Password",
            "type": "password",
            "value": "password",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "login-input",
            "name": "passwordTwo",
            "onChange": [Function],
            "placeholder": "Retype password",
            "type": "password",
            "value": "password",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
